.github/ci-scripts/bash/functions-for-build

```diff
@@ -565,6 +575,64 @@ setup_qa_env() {
   deactivate
 }
 
+setup_packer_env() {
+  sudo python3 -m pip uninstall -y packer_framework
+  python3 -m pip install --user virtualenv --trusted-host ${NEXUS_SERVER_ENDPOINT}
+  python3 -m venv pkenv
+  source pkenv/bin/activate
+  python3 -m pip install wheel --trusted-host ${NEXUS_SERVER_ENDPOINT}
+  python3 -m pip install -r ./requirements.txt --extra-index-url=https://$GITHUB_USER:$GITHUB_TOKEN@${NEXUS_SERVER_ENDPOINT}/repository/${NEXUS_PULL_REPOS_PY}/simple
+
+  find /usr/bin /usr/local/bin -name build_all_artifacts
+  rm -rf /usr/lib/python3/dist-packages/OpenSSL
+  pip3 install pyopenssl
+  pip3 install pyopenssl --upgrade
+  pip3 install --force-reinstall setuptools==45.2.0
+}
+
+software_def_build() {
+  setupPackerEnv
+  source pkenv/bin/activate
+  mkdir -p ~/.packer.d/plugins
+  if [[ "$BRANCH_NAME" == *double_commit* ]]; then
+    echo "BRANCH_NAME contains double_commit"
+    get_pr_commits
+    if [[ "$PR_COMMITS" -le 2 ]]; then
+      echo "BRANCH_NAME contains double_commit and less than 2 commits, skipping build"
+      return
+    fi
+  fi
+  validate_all -r $PWD --src_branch ${SOURCE_BRANCH} --pipeline_run
+  build_all_artifacts -r $PWD --dest_branch ${GITHUB_BASE_REF} --src_branch ${SOURCE_BRANCH} --dry_run --pipeline_run
+}
+
+software_def_build_and_push() {
+  setupPackerEnv
+  export CURRENT_VERSION_TAG=${LATEST_TAG}
+  source pkenv/bin/activate
+  echo $CURRENT_VERSION_TAG
+  mkdir -p ~/.packer.d/plugins
+  if [[ "$BRANCH_NAME" == *double_commit* ]]; then
+    echo "BRANCH_NAME contains double_commit"
+    get_pr_commits
+    if [[ "$PR_COMMITS" -le 2 ]]; then
+      major=$(echo "$LATEST_TAG" | cut -d. -f1)
+      prev_minor=$(($(echo "$LATEST_TAG" | cut -d. -f2) - 1))
+      PREVIOUS_LATEST_TAG=$(git tag | grep "$major.$prev_minor." | tail -n 1)
+      PREVIOUS_LABEL="rel-$major.$prev_minor.0"
+      URL_PATH="${NEXUS_ARTIFACT_REPO}/${PREVIOUS_LABEL}/delta-image-manifest/delta-image-manifest-${PREVIOUS_LATEST_TAG}.json"
+      echo "${NEXUS_PROTOCOL}://${NEXUS_SERVER_ENDPOINT}/repository/${URL_PATH}"
+      curl -o common_manifest.json --user "${GITHUB_USER}:${GITHUB_TOKEN}" \
```

Comment:
Consider a case where:
Production - rel-0.122.0
QA/RC - rel-0.123.0
Dev - main

And a PR merged on `rel-0.122.0`; 
In which case the DC PR of main, will try to get the `delta-image-manifest-<tag>.json` of rel-0.123.0`;  And we might not get the right tag

Its always good to find the release of branch where actual PR was merged and use that tag